instrumentation/spring/spring-webmvc/spring-webmvc-3.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/webmvc/v3_1/DispatcherServletInstrumentation.java

```diff
@@ -76,29 +77,40 @@ public static void afterRefresh(
   @SuppressWarnings("unused")
   public static class RenderAdvice {
 
+    public static class AdviceScope {
+      private final Context context;
+      private final Scope scope;
+
+      public AdviceScope(Context context, Scope scope) {
+        this.context = context;
+        this.scope = scope;
+      }
+
+      public void exit(ModelAndView mv, @Nullable Throwable throwable) {
+        scope.close();
+        modelAndViewInstrumenter().end(context, mv, null, throwable);
+      }
+    }
+
+    @Nullable
     @Advice.OnMethodEnter(suppress = Throwable.class)
-    public static void onEnter(
-        @Advice.Argument(0) ModelAndView mv,
-        @Advice.Local("otelContext") Context context,
-        @Advice.Local("otelScope") Scope scope) {
+    public static AdviceScope onEnter(@Advice.Argument(0) ModelAndView mv) {
       Context parentContext = currentContext();
-      if (modelAndViewInstrumenter().shouldStart(parentContext, mv)) {
-        context = modelAndViewInstrumenter().start(parentContext, mv);
-        scope = context.makeCurrent();
+      if (!modelAndViewInstrumenter().shouldStart(parentContext, mv)) {
```

Comment:
Same here: why not move the logic into a `static enter()` method on `AdviceScope`?